packages/framework/src/Model/Complaint/Complaint.php

```diff
@@ -0,0 +1,295 @@
+     * @ORM\OneToMany(
+     *     targetEntity="Shopsys\FrameworkBundle\Model\Complaint\ComplaintItem",
+     *     mappedBy="complaint",
+     *     cascade={"persist"},
+     *     orphanRemoval=true
+     * )
+     */
+    protected $items;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Complaint\ComplaintData $complaintData
+     * @param \Shopsys\FrameworkBundle\Model\Complaint\ComplaintItem[] $complaintItems
+     */
+    public function __construct(ComplaintData $complaintData, array $complaintItems)
+    {
+        $this->createdAt = new DateTime();
+        $this->setItems($complaintItems);
+
+        $this->setData($complaintData);
+    }
+
+    /**
+     * @return int
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return string
+     */
+    public function getUuid()
+    {
+        return $this->uuid;
+    }
+
+    /**
+     * @return string
+     */
+    public function getNumber()
+    {
+        return $this->number;
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Model\Order\Order
+     */
+    public function getOrder()
+    {
+        return $this->order;
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser|null
+     */
+    public function getCustomerUser()
+    {
+        return $this->customerUser;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getDeliveryFirstName()
+    {
+        return $this->deliveryFirstName;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getDeliveryLastName()
+    {
+        return $this->deliveryLastName;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getDeliveryCompanyName()
+    {
+        return $this->deliveryCompanyName;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getDeliveryTelephone()
+    {
+        return $this->deliveryTelephone;
+    }
+
+    /**
+     * @return string
+     */
+    public function getDeliveryStreet()
+    {
+        return $this->deliveryStreet;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getDeliveryCity()
+    {
+        return $this->deliveryCity;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getDeliveryPostcode()
+    {
+        return $this->deliveryPostcode;
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Model\Country\Country|null
+     */
+    public function getDeliveryCountry()
+    {
+        return $this->deliveryCountry;
+    }
+
+    /**
+     * @return \DateTime
+     */
+    public function getCreatedAt()
+    {
+        return $this->createdAt;
+    }
+
+    /**
+     * @return string
+     */
+    public function getStatus()
+    {
+        return $this->status;
+    }
+
+    /**
+     * @return \Doctrine\Common\Collections\ArrayCollection|\Doctrine\Common\Collections\Collection
+     */
+    public function getItems()
+    {
+        return $this->items;
```

Comment:
collections should not be exposed to outside entities to avoid unforeseen consequences while modifying the data.

```suggestion
        return $this->items->getValues();
```

Reply:
Fixed [fixup! complaint data structure](https://github.com/shopsys/shopsys/pull/3295/commits/4217bb1d28c8d1dc65cee853972b462153ce868b)